spec/models/customer_spec.rb

```diff
@@ -6,8 +6,17 @@
   end
 
   it 'counts how many successful transactions a customer has' do
-    customer = Customer.first
+    customer = create(:customer)
+    invoice1 = create(:invoice, customer: customer)
+    invoice2 = create(:invoice, customer: customer)
+    invoice3 = create(:invoice, customer: customer)
 
-    expect(customer.successful_transactions_count).to eq(7)
+    create(:transaction, result: 'success', invoice: invoice1)
+    create(:transaction, result: 'success', invoice: invoice1)
+    create(:transaction, result: 'success', invoice: invoice2)
+    create(:transaction, result: 'failed', invoice: invoice2)
+    create(:transaction, result: 'failed', invoice: invoice3)
+
+    expect(customer.successful_transactions_count).to eq(3)
```

Comment:
I like that you included multiple failing transactions with different invoices, good testing imo